service/frontend/nexus_handler.go

```diff
@@ -90,7 +91,7 @@ type operationContext struct {
 	telemetryInterceptor          *interceptor.TelemetryInterceptor
 	redirectionInterceptor        *interceptor.Redirection
 	forwardingEnabledForNamespace dynamicconfig.BoolPropertyFnWithNamespaceFilter
-	cleanupFunctions              []func(error)
+	cleanupFunctions              []func(map[string]string, error)
```

Comment:
Maybe accept the context here instead? Not critical, we're going to refactor all of this.

Reply:
I thought about that but figured it was a bit overkill and yeah we're going to change it anyway.